coconut/src/proof.rs

```diff
@@ -210,12 +219,12 @@ impl RequestCoinsProof {
             CoconutError::MalformedProof
         );
         ensure!(
-            parameters.max_attributes() >= 2,
-            CoconutError::MalformedProof
+            parameters.max_attributes() >= 3,
+            CoconutError::TooManyAttributes
```

Comment:
Mmm, this is a bit confusing. I think what you mean is that the setup did not prepare enough parameters to support 3 attributes?

Reply:
exactly. I mean that we are requesting creds  on too many attributes

Reply:
do you have a better name in mind? (happy to change)

Reply:
Yeah, it's fine. Perhaps this is slightly better, surprisingly: `ensure!(3 <= public_key.max_attributes(), CoconutError::TooManyAttributes)`